### ISSUE TITLE
Make map markers clickable

### ISSUE THREAD

**Issue by atruskie** [Allcharles/workbench-client#2026]
**Is your feature request related to a problem? Please describe.**

When users look at maps it would be nice if the could click though to the items indicated.

**Describe the solution you'd like**

Make map markers clickable.

- [ ] On the site/region page
- [ ] Survey for any other places the maps are used

**Describe alternatives you've considered**

N/A

**Additional context**

User requested feature.

**Comment by hudson-newey** [comment]
This was previously on hold because it required some architectural changes to the `map.component` to register / emit click events.

The required architectural work will completed in https://github.com/QutEcoacoustics/workbench-client/pull/2462

Once this PR is merged, we should be able to just listen for the `markerClicked` event on the `site-map` component
